build/patches/experimental-user-scripts-support.patch

```diff
@@ -0,0 +1,10299 @@
++    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
++    GNU General Public License for more details.
++
++    You should have received a copy of the GNU General Public License
++    along with Bromite. If not, see <https://www.gnu.org/licenses/>.
++*/
++
++#ifndef COMPONENTS_USERSCRIPS_HELPER_ANDROID_H_
++#define COMPONENTS_USERSCRIPS_HELPER_ANDROID_H_
++
++#include <jni.h>
++
++namespace user_scripts {
++
++static void ShouldRefreshUserScriptList(JNIEnv* env);
++static void OnUserScriptLoaded(JNIEnv* env,
++              bool result, const std::string& error);
++
++}  // namespace user_scripts
++
++#endif
+\ No newline at end of file
+diff --git a/components/user_scripts/browser/BUILD.gn b/components/user_scripts/browser/BUILD.gn
+new file mode 100755
+--- /dev/null
++++ b/components/user_scripts/browser/BUILD.gn
+@@ -0,0 +1,91 @@
++# This file is part of Bromite.
++
++# Bromite is free software: you can redistribute it and/or modify
++# it under the terms of the GNU General Public License as published by
++# the Free Software Foundation, either version 3 of the License, or
++# (at your option) any later version.
++
++# Bromite is distributed in the hope that it will be useful,
++# but WITHOUT ANY WARRANTY; without even the implied warranty of
++# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
++# GNU General Public License for more details.
++
++# You should have received a copy of the GNU General Public License
++# along with Bromite. If not, see <https://www.gnu.org/licenses/>.
++
++import("//build/config/features.gni")
++import("//tools/grit/grit_rule.gni")
++
++group("browser") {
++  public_deps = [
++    "//components/user_scripts/browser:browser_sources",
++  ]
++}
++
++source_set("browser_sources") {
++  visibility = [ "./*" ]
++
++  sources = [
++    "file_task_runner.cc",
++    "file_task_runner.h",
++    "userscripts_browser_client.cc",
++    "userscripts_browser_client.h",
++    "user_script_loader.cc",
++    "user_script_loader.h",
++    "user_script_prefs.cc",
++    "user_script_prefs.h",
++    "user_script_pref_info.cc",
++    "user_script_pref_info.h",
++    "ui/user_scripts_ui.h",
++    "ui/user_scripts_ui.cc",
++  ]
++
++  deps = [
++    ":userscripts_browser_resources",
++    "//base:i18n",
++    "//components/keyed_service/content",
++    "//components/keyed_service/core",
++    "//components/pref_registry",
++    "//components/prefs",
++    "//content/public/browser",
++    "//crypto:platform",
++    "//components/user_scripts/common",
++    "//services/device/public/mojom",
++    "//services/preferences/public/cpp",
++    "//services/service_manager/public/cpp",
++    "//third_party/blink/public/common",
++    "//third_party/blink/public/mojom/frame",
++    "//ui/display",
++  ]
++
++  public_deps = [
++    "//content/public/common",
++  ]
++
++  configs += [
++    "//build/config:precompiled_headers",
++
++    # TODO(jschuh): crbug.com/167187 fix size_t to int truncations.
```

Comment:
Are we affected also by these? Try building without? Otherwise this TODO will go stale since nobody from upstream will remove it from our patch.

Reply:
no problem,
when https://source.chromium.org/chromium/chromium/src/+/main:build/config/compiler/BUILD.gn;l=1931 disappears (and we will have an error on the gn) we remove it too